samples/snippets/core/system-text-json/csharp/Utf8ReaderPartialRead.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+
+namespace SystemTextJsonSamples
+{
+    public class Utf8ReaderPartialRead
+    {
+        public static void Run()
+        {
+            var jsonString = @"{
+                ""Date"": ""2019-08-01T00:00:00-07:00"",
+                ""Temperature"": 25,
+                ""TemperatureRanges"": {
+                    ""Cold"": { ""High"": 20, ""Low"": -10 },
+                    ""Hot"": { ""High"": 60, ""Low"": 20 }
+                },
+                ""Summary"": ""Hot"",
+            }";
+
+            byte[] bytes = Encoding.UTF8.GetBytes(jsonString);
+            var stream = new MemoryStream(bytes);
+
+            var buffer = new byte[4096];
+
+            // Fill the buffer.
+            // For this snippet, we're assuming the stream is open and has data.
+            // If it might be closed or empty, check if the return value is 0.
+            stream.Read(buffer);
+
+            var reader = new Utf8JsonReader(buffer, isFinalBlock: false, state: default);
```

Comment:
I am curious, does this sample successfully run when the buffer is larger than the payload, i.e. 4096. I would have thought that `isFinalBlock: false` when the entire JSON is provided in one chunk would have caused an exception.

Reply:
It doesn't cause an exception, it works fine at 4096.